lib/puppetserver/ca/action/create.rb

```diff
@@ -152,9 +153,14 @@ def generate_key_csr(certname, settings, digest)
           private_key = host.create_private_key(settings[:keylength])
           extensions = []
           if !settings[:subject_alt_names].empty?
-            extensions << host.create_extension("subjectAltName", settings[:subject_alt_names])
+            s_value = OpenSSL::ASN1::UTF8String.new(settings[:subject_alt_names].to_s)
+            extensions << OpenSSL::X509::Extension.new("subjectAltName", s_value.to_der, false)
```

Comment:
So turns out for whatever reason you don't want to process subject alt names this way. The server throws an error down in the Java signing code if you convert this to a UTF8 string DER. Putting this back to what you had before (I didn't add the function back, just put its contents here) makes it work again.